### PR TITLE
Modify S4426: Learn-As-You-Code Migration

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S4426/common/fix/fix.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+A larger key size increases the number of possible keys and makes exhaustive
+search attacks computationally infeasible. Therefore, a 256-bit key provides a
+higher level of security than a 128-bit or 192-bit key.
+
+Currently, a minimum key size of *192 bits* is recommended for AES.
+
+==== Elliptic Curve Cryptography (ECC)
+
+Elliptic curve cryptography is also used in various algorithms, such as ECDSA,
+ECDH, or ECMQV. Keys generated with EC algorithms are mentioned directly in
```
Comment (loris-s-sonarsource) [inline rules/S4426/common/fix/fix.adoc:28]:
```suggestion
ECDH, or ECMQV. The length of keys generated with EC algorithms are mentioned directly in
```

File: rules/S4426/impact.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+=== What is the potential impact?
+After retrieving encrypted data and performing mathematical attacks on it on a
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S4426/impact.adoc:2]: I am unsure if "mathematical attacks" is a good description. Maybe "exhaustion attack"? https://www.staff.uni-mainz.de/pommeren/Cryptology/Classic/1_Monoalph/exhaust.html

File: rules/S4426/common/fix/fix.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
+As a rule of thumb, use the cryptographic algorithms and mechanisms that are
+considered strong by the cryptographic community. It is of course also always
+possible to use higher key sizes that provide an even higher level of security.
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S4426/common/fix/fix.adoc:3]: I have to admit that when I wrote that I felt some kind of imprecision

File: rules/S4426/kotlin/how-to-fix-it/java-cryptographic-extension.adoc [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+include::../../common/fix/rsa.adoc[]
+
+[source,kotlin,diff-id=1,diff-type=noncompliant]
+----
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+
+public static void main(String[] args) {
+    try {
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
```
Comment (loris-s-sonarsource) [inline rules/S4426/kotlin/how-to-fix-it/java-cryptographic-extension.adoc:18]:
```suggestion
        val keyPairGenerator = KeyPairGenerator.getInstance("RSA");
```

File: rules/S4426/kotlin/how-to-fix-it/java-cryptographic-extension.adoc [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+include::../../common/fix/aes.adoc[]
+
+[source,kotlin,diff-id=2,diff-type=noncompliant]
+----
+import java.security.KeyGenerator;
+import java.security.NoSuchAlgorithmException;
+
+public static void main(String[] args) {
+    try {
+        KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
```
Comment (loris-s-sonarsource) [inline rules/S4426/kotlin/how-to-fix-it/java-cryptographic-extension.adoc:36]:
```suggestion
        val keyGenerator = KeyGenerator.getInstance("AES");
```

File: rules/S4426/java/rule.adoc [hunk 6]
```diff
@@ -1,38 +1,30 @@
...
+include::../common/resources/presentations.adoc[]
 
-KeyPairGenerator keyPairGen5 = KeyPairGenerator.getInstance("EC");
-ECGenParameterSpec ecSpec10 = new ECGenParameterSpec("secp256r1"); // compliant
-keyPairGen5.initialize(ecSpec10); 
+include::../common/resources/standards.adoc[]
+* https://mobile-security.gitbook.io/masvs/security-requirements/0x08-v3-cryptography_verification_requirements[Mobile AppSec Verification Standard] - Cryptography Requirements
+* https://owasp.org/www-project-mobile-top-10/2016-risks/m5-insufficient-cryptography[OWASP Mobile Top 10 2016 Category M5] - Insufficient Cryptography
+* https://cwe.mitre.org/data/definitions/327[MITRE, CWE-327] - Use of a Broken or Risky Cryptographic Algorithm
+* https://wiki.sei.cmu.edu/confluence/x/hDdGBQ[CERT, MSC61-J.] - Do not use insecure or weak cryptographic algorithms
```
Comment (loris-s-sonarsource) [inline rules/S4426/java/rule.adoc:25]:
```suggestion
```

File: rules/S4426/python/rule.adoc [hunk 7]
```diff
@@ -1,66 +1,37 @@
...
 == Implementation Specification
 (visible only on this page)
 
-=== Message
-
-Use a key length of at least 'xxx' bits
-
-Use a public key exponent of at least 65537
-
+include::../message.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S4426/python/rule.adoc:29]:
```suggestion
i=== Message

Use a key length of at least 'xxx' bits

Use a public key exponent of at least 65537
```

**Review by hendrik-buchwald-sonarsource — CHANGES_REQUESTED** [review]
`validate_asciidoc` is still failing, otherwise it looks good.

**Pull request merged**